services/wallet/service.go

```diff
@@ -689,6 +719,87 @@ func (service *Service) LinkUpholdWallet(ctx context.Context, wallet uphold.Wall
 	return country, nil
 }
 
+func (service *Service) LinkSolanaAddress(ctx context.Context, paymentID uuid.UUID, req linkSolanaAddrRequest) error {
+	if err := isWalletWhitelisted(ctx, service.Datastore.RawDB(), service.allowListRepo, paymentID); err != nil {
+		return err
+	}
+
+	ctx, txn, rollback, commit, err := getTx(ctx, service.Datastore)
+	if err != nil {
+		return err
+	}
+	defer rollback()
+
+	chl, err := service.chlRepo.Get(ctx, txn, paymentID.String())
+	if err != nil {
+		return err
+	}
+
+	if err := chl.IsValid(time.Now()); err != nil {
+		return err
+	}
+
+	w, err := service.Datastore.GetWallet(ctx, paymentID)
+	if err != nil {
+		return err
+	}
+
+	if w == nil {
+		return model.ErrWalletNotFound
+	}
+
+	if err := w.LinkSolanaAddress(ctx, walletutils.SolanaLinkReq{
+		Pub:   req.SolanaPublicKey,
+		Sig:   req.SolanaSignature,
+		Msg:   req.Message,
+		Nonce: chl.Nonce,
+	}); err != nil {
+		return err
+	}
+
+	cl := NewSolanaCustodialLink(paymentID, w.UserDepositDestination)
+
+	if err := service.Datastore.LinkWallet(ctx, w.ID, w.UserDepositDestination, *cl.LinkingID, cl.Custodian); err != nil {
+		return err
+	}
+
+	if err := service.chlRepo.Delete(ctx, txn, chl.ID); err != nil {
+		return err
+	}
+
+	if err := commit(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (service *Service) linkCustodialAccount(ctx context.Context, wID string, userDepositDestination string, providerLinkingID uuid.UUID, depositProvider, country string) error {
+	walletID, err := uuid.FromString(wID)
```

Comment:
Going back to the question about passing UUIDs, this is another example of such situation.

Reply:
The problem is the [wallet model](https://github.com/brave-intl/bat-go/blob/9a45bf9d2890829f506fa8a4c434f83f07f35824/libs/wallet/wallet.go#L17) uses a string for `ID` when it should be a `UUID` which causes us to switch between the two as we use a `UUID` as the paymentID pretty much everywhere else. As part of this PR I have isolated the conversion in this function and in the subsequent wallet refactor this will be removed and the wallet model updated to use a `UUID`, this will allow us to remove all the string to uuid conversions when dealing with the wallet. 

There is a separate issue to deal with this as its too large and not relevant to this issue.